synapse/tests/test_tools_deploy.py

```diff
@@ -0,0 +1,60 @@
+import synapse.common as s_common
+
+import synapse.tests.common as s_test
+
+import synapse.tools.deploy as s_deploy
+
+class DeployTest(s_test.SynTest):
+    def test_deploy_cells(self):
+        outp = self.getTestOutp()
+        argv = ['--cells', 'hehe', 'haha']
+        ret = s_deploy.main(argv, outp)
+        self.eq(ret, 0)
+        outp.expect('Registered cells:')
+        outp.expect('cortex')
+
+    def test_deploy_auth(self):
+        with self.getTestDir() as dirn:
+            outp = self.getTestOutp()
+            argv = ['cortex', 'core', dirn, '--auth']
+            ret = s_deploy.main(argv, outp)
+            self.eq(ret, 0)
+            d = s_common.yamlload(dirn, 'cells', 'core', 'boot.yaml')
+            self.eq(d, {'auth:en': True,
+                        'type': 'cortex',
+                        'cell:name': 'core'})
+            # Sad path
+            outp = self.getTestOutp()
+            argv = ['cortex', 'core', dirn, '--auth']
+            ret = s_deploy.main(argv, outp)
+            self.eq(ret, 1)
+            outp.expect('cell directory already exists')
+
+        with self.getTestDir() as dirn:
+            outp = self.getTestOutp()
+            argv = ['cortex', 'core', dirn, '--admin', 'pennywise:clownshoes']
+            ret = s_deploy.main(argv, outp)
+            self.eq(ret, 0)
+            d = s_common.yamlload(dirn, 'cells', 'core', 'boot.yaml')
+            self.eq(d, {'auth:en': True,
+                        'type': 'cortex',
+                        'cell:name': 'core',
+                        'auth:admin': 'pennywise:clownshoes'})
+
+    def test_deploy_dmonyaml(self):
+        with self.getTestDir() as dirn:
+            outp = self.getTestOutp()
+            argv = ['--listen', 'tcp://1.2.3.4:8080/', '--module', 'synapse.tests.utils', 'cortex', 'core', dirn]
```

Comment:
We don't have to change anything in this PR, but we might want to have `--listen` validate its arg before saving so people don't type `0.0.0.0:80` and have it blow up later when there's no proto